File: src/AspCoreTest/Models/ShipmentViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace AspCoreTest.Models
{
    public class ShipmentViewModel
    {
        public int id { get; set; }
        public string from { get; set; }
        public string to { get; set; }
        public DateTime dateTimeOut { get; set; }
        public DateTime dateTimeInput { get; set; }
        public List<int> places { get; set; }
        public decimal price { get; set; }
    }
}

File: src/AspCoreTest/Scripts/ShipmentViewModel.ts
export class Shipment {
    constructor(
        public id: number,
        public from: string,
        public To: string,
        public dateTimeOut: any,
        public dateTimeInput: any,
        public places: Array<number>,
        public price: number)
    { }
}

export class Direction {
    constructor(
        public from: string,
        public to: string
    )
    { }
}